src/components/JobsPanel/jobsPanel.util.js

```diff
@@ -216,18 +237,20 @@ export const generateTableData = (
             value: env.value ?? ''
           }
         })),
-        secretSources: []
+        secretSources: [],
+        node_selector
       }
     })
     setNewJob({
       inputs: parseDefaultDataInputsContent(dataInputs),
       parameters: parseDefaultContent(parameters),
-      volume_mounts: volumeMounts.length
-        ? volumeMounts.map(volumeMounts => volumeMounts.data)
-        : [],
+      volume_mounts: (volumeMounts || []).map(
```

Comment:
Doesn't `??` suffice here? 🤔 
```suggestion
      volume_mounts: (volumeMounts ?? []).map(
```